packages/admin-next/dashboard/src/lib/form-helpers.ts

```diff
@@ -0,0 +1,25 @@
+import { castNumber } from "./cast-number"
+
+export function parseOptionalFormValue<T>(value: T): T | undefined {
+  if (typeof value === "string" && value.trim() === "") {
```

Comment:
Why would you want to set anything to an empty string? Don't think that makes sense from a form input perspective, as an empty string equals undefined or null in practice as you haven't input anything.

Reply:
Undefined is stripped from the json once sent no? and for us, undefined means a property as not been provided if i am right. In general for the value we look for null, empty and value cc @olivermrbl

Reply:
You maybe want to set a product subtitle to an empty string (or in other words, unset it). With this approach, that won't be possible.

What we can do is check if the initial value differs from the current one (i.e initially it wasn't empty, but it got set to empty when submitting).

Reply:
If you want to unset the subtitle you should send null to the backend. I have just updated the PR to also allow passing null for all the nullable fields, as well as this util so it now transforms empty string into null.

Reply:
@sradevski if it's a create form, you properly don't want to nullify the fields, but instead convert them to `undefined` so the values aren't sent off to the backend. But it does not really matter, as the fields that are nullable, are also null by default in our models. But in most cases you would set nullify to `false` here. In edit forms, you want to send null because an empty field means you want to remove the value. We could also check the diff between initial values and current values, but I don't see the point in adding additional code for this, when the above pattern is always true. But maybe I am missing something, but my opinions is that:

1. Our validators should accept `null` for all fields that are nullable, in order to allow users to correctly unset the value, and not have to set empty strings for all of their fields, which also does not work for number fields that should be nullable.
2. In edit forms an empty input === null, in create forms an empty input === undefined.

Reply:
@kasperkristensen The only concern I had is in your second point, on edit an empty input === null, it could be that we want an empty string instead. But maybe it's an edge case so le's try this and see if we stumble upon issues.